### DISCUSSION TITLE
Input with icon

### DISCUSSION THREAD

**Discussion by AL-THE-BOT-FATHER** [zauberzeug/nicegui#409] [Q&A] [answered]
![search](https://user-images.githubusercontent.com/98119917/220130300-2bce973b-80ae-4aff-818c-b485d3d17c44.PNG)

```py
from nicegui import ui
with ui.card().classes("m-auto").style("width: 500px"):
    with ui.row().classes('items-center'):
        search_field = ui.input(label="Search").props('autofocus outlined rounded item-aligned').classes('w-96 self-center')
        ui.icon('search').props("size=3rem")
        
ui.run(title="Test", dark=True)
```

How do I get the search icon in the input field? 

Thanks. 

-Allen

**Comment by rodja** [comment]
This could be related to #300?

**Comment by Diegiwg** [comment]
The input has a 'slot' called append, which does the effect you are looking for. The code would be:

```python
from nicegui import ui

icon = ui.icon('search')

with ui.card().classes("m-auto").style("width: 500px"):
    with ui.row().classes('items-center'):
        search_field = ui.input(label="Search")
            
        append_slot = search_field.add_slot("append")
        append_slot.children.append(icon)
ui.run()
```
The problem: Slot only accepts a list of elements as children to be rendered.

However, I couldn't find a way to pass an element to be rendered only within the context of the slot.

Elements are always rendered in-context (correct behavior), and out-of-context (incorrect behavior).

So, in the example above, you will have the icon correctly inside the input, however, the same icon will appear on the page, as a normal element.

**Reply by falkoschindler** [reply]
See https://github.com/zauberzeug/nicegui/issues/300#issuecomment-1405769302 for an example of how to place an icon in the"append" slot. 🙂

**Comment by Diegiwg** [comment, marked as answer]
I didn't know how to use the Slots, an interesting knowledge, since Quasar has several Slots.

Well, thanks to @falkoschindler, now the icon repeating problem doesn't happen anymore.

Here is an example working perfectly:

```python
from nicegui import ui

with ui.card().classes("m-auto").style("width: 500px"):
    with ui.row().classes('items-center'):
        search_field = ui.input(label="Search")
            
        with search_field.add_slot("append"):
            icon = ui.icon('search')
ui.run()
```
![working example](https://user-images.githubusercontent.com/15692310/220357717-ff07552b-059a-41a4-a70e-67f480577597.PNG)

**Reply by AL-THE-BOT-FATHER** [reply]
Thanks man. Looks good. 

![image](https://user-images.githubusercontent.com/98119917/220387868-eeee4edd-6ffc-4b68-b3eb-7726505d4a9b.png)
